### PR TITLE
Mobile Implemantation: Filtering issue by Catagory and Issue Detail Bloc

### PR DESCRIPTION
**Pull request by M-A-D-A-R-A**
# Description
 - Filtering the issue by selected category
 - Created Bloc for updating and deleting the issue 
 - Issue Detail UI (create an issue. delete issue)
 - Comment Feature (Post,update,delete comment)

https://user-images.githubusercontent.com/56160262/185641634-175bdd13-8eb2-4410-bd96-79139d95fd85.mp4

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

*Test Configuration*:

- Firmware version: Windows
- Hardware: SM M426B
- SDK: Flutter SDK (v 3.0.2)

Also, include screenshots of the app for verification and reviewing purposes.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by amlannandy — CHANGES_REQUESTED** [review]
Can you share the console logs

**Comment by amlannandy** [comment]
@M-A-D-A-R-A Check the `getIssue` method in `labellab_api_impl.dart`.
Specifically this part -

```
Issue issue = Issue.fromJson(
    response.data['body']['items'],
);
```

This is not a list, just a single item. So the data will be in `response.data['body']` not in `response.data['body']['items']`.
Please pay attention when writing these code snippets, don't straight-up copy the code from the listing issues method.

**Comment by amlannandy** [comment]
@M-A-D-A-R-A Remove unnecessary print statements from the code.
Printing errors is fine, but not random statements like this.

`add_edit_issue_screen.dart` - Line 214

```
print(issue.description);
```

**Comment by amlannandy** [comment]
@M-A-D-A-R-A Also please self-review the changes while creating the PR. A lot of these small mistakes won't come up if you self-evaluate a bit.

File: labellab_mobile/lib/screen/issue/details/issue_details_state.dart
Comment (amlannandy) [inline labellab_mobile/lib/screen/issue/details/issue_details_state.dart:10]: Why are we passing the issue as argument here. Not necessary. We only have to toggle the loading state here.

**Comment by M-A-D-A-R-A** [comment]
@amlannandy please review.

**Review by amlannandy — CHANGES_REQUESTED** [review]
The UI doesn't look good at all. The font sizes and spacing are all mismatched. Please review the rest of the application and see how we have implemented it. I've already given instructions on the Figma and during meets. Please go through them carefully and make the fixes.
I can't merge this kind of work.

**Comment by M-A-D-A-R-A** [comment]
Made the necessary changes(Reference video in Description), Please review, Also would it be okay if we merge this PR after making the Necessary changes in the Issue Details Screen?

File: labellab_mobile/lib/routing/route_handler.dart
Comment (amlannandy) [inline labellab_mobile/lib/routing/route_handler.dart:51]: In case you didn't know, absolute import means something like this -

```
import 'package:labellab_mobile/screen/issue/issue_activity/issue_activity_screen.dart';

Please fix these
````

File: labellab_mobile/lib/screen/issue/add_edit/add_edit_issue_screen.dart
Comment (amlannandy) [inline labellab_mobile/lib/screen/issue/add_edit/add_edit_issue_screen.dart:216]: Remove this please

**Review by amlannandy — CHANGES_REQUESTED** [review]
There are still many imports and unnecessary logs and comments that you haven't removed.
PLEASE go through your changes CAREFULLY and make the corrections.
Remove the changes for Issue Tile UI changes from this PR.

**Comment by M-A-D-A-R-A** [comment]
@amlannandy Please Review.

**Review by amlannandy — APPROVED** [review]
LGTM. Nice work 👍🏻

**Comment by amlannandy** [comment]
@M-A-D-A-R-A Can you rebase your branch with upstream/main ?

**Comment by M-A-D-A-R-A** [comment]
> @M-A-D-A-R-A Can you rebase your branch with upstream/main ?

Yeah sure!

**Comment by M-A-D-A-R-A** [comment]
@amlannandy can u please check if there are still merge conflicts?

**Review by amlannandy — DISMISSED** [review]

**Comment by amlannandy** [comment]
@M-A-D-A-R-A Nope. Looks like you didn't rebase it properly. There are a lot of extra commits in the PR that shouldn't be there.

1. Add scorelab/LabelLab as `upstream`
2. `git fetch upstream`
3. `git checkout M-A-D-A-R-A:sprint6_LL32`
4. `git rebase upstream/master`

Run these

**Review by amlannandy — APPROVED** [review]

**Comment by amlannandy** [comment]
@M-A-D-A-R-A Still can't merge. Can you try creating a new branch and cherry-picking these commits onto them one by one? Should work.

**Pull request closed without merging**
